File: 941 - Valid Mountain Array.py
class Solution:
    def validMountainArray(self, arr: list[int]) -> bool:
        n = len(arr)
        sinc = False
        sdec = False
        if n < 3: 
            return False
        for i in range(1, n):
            if arr[i] > arr[i - 1]:
                if not sinc and not sdec:
                    sinc = True
                elif sinc and sdec:
                    return False
            elif arr[i] < arr[i - 1]:
                if sinc and not sdec:
                    sdec = True
                elif sinc and sdec:
                    continue
                else:
                    return False
            else:
                return False
        return sinc and sdec

File: 941 - Valid Mountain Array.cpp
class Solution {
public:
    bool validMountainArray(vector<int>& arr) {
        bool sinc = false, sdec = false;
        int n = arr.size();
        if (n < 3) return false;
        for (int i = 1; i < n; i++) {
            if (arr[i] > arr[i - 1]) {
                if (!sinc && !sdec) {
                    sinc = true;
                } else if (sinc && sdec) {
                    return false;
                }
            } else if (arr[i] < arr[i - 1]) {
                if (sinc && !sdec) {
                    sdec = true;
                } else if (sinc && sdec) {
                    continue;
                } else {
                    return false;
                }
            } else {
                return false;
            }
        }
        return sinc && sdec;
    }
};